docs/docs/how-to/dbauth-passwordless.md

```diff
@@ -0,0 +1,644 @@
+
+One thing though is now you're collecting the user's login, password and salt.  If you'd like to not collect all of this, an alternative is passwordless.  The only data needed for passwordless is the users email address.
+
+In this how-to I'll show you how to set up dbAuth to be passwordless, you'll still need to set up a a way to send [send emails](../how-to/sending-emails.md), but there's plenty of ways to do that.
+
+## Background
+
+Let me start by sharing a little bit about how passwordless works.
+### What is a passwordless authentication method?
+A passwordless authentication method is a method of authentication where the user is not required to enter a password.  Instead, the user is sent a link to their email address.  When they click the link, they are logged in.
+
+Passwordless uses a token that is time-sensitive.  So instead of storing a password, we store a token, and an expiration.
+
+That token is generated randomly and is stored in the database.
+## How to do it
+### 1. Modify the Prisma schema
+First, we need to modify the Prisma schema.
+
+If you followed the tutorial you'll have a `User` model.  Here's is what it looks like with after the changes.
+
+```jsx {4-6}
+model User {
+  id                  Int       @id @default(autoincrement())
+  name                String?
+  email               String    @unique
+  loginToken          String
+  loginTokenExpiresAt DateTime?
+  salt                String?
+}
+```
+
+Make note of the optional `salt` field.
+
+Once you've made the changes, you'll need to migrate your database.
+
+```bash
+yarn rw prisma migrate dev
+```
+
+### 2. Setting up the generateToken function
+Next, we need to create a function that will generate a token and an expiration date.
+
+If you followed the tutorial, you might not have a `/api/src/services/users/users.js` file.  If that's the case, you can create it.
+
+```bash
+yarn rw g service users
+```
+
+Now that you have the file, let's add the `generateToken` function.
+
+```javascript {21} title="/api/src/services/users/users.js"
+// add this to the bottom of the file
+export const generateLoginToken = async ({ email }) => {
+  try {
+    // look up if the user exists
+    let lookupUser = await db.user.findFirst({ where: { email } })
+    if (!lookupUser) return { message: 'Login Request received' }
+    // here we're going to generate a random password of 6 numbers
+    let randomNumber = (() => {
+      let random = CryptoJS.lib.WordArray.random(6)
+      let randomString = random.toString()
+      let sixDigitNumber = randomString.replace(/\D/g, '')
+      if (sixDigitNumber.length < 6) {
+        sixDigitNumber = sixDigitNumber.padStart(6, '0')
+      }
+      if (sixDigitNumber.length > 6) {
+        sixDigitNumber = sixDigitNumber.slice(0, 6)
+      }
+      return sixDigitNumber.toString()
+    })()
+    console.log({ randomNumber }) // email the user this number
```

Comment:
I'm not sure if this is a great way to do this, if there's a better way I'm all ears.  It works.

Reply:
hmm, what does `WordArray` return, letters and numbers? Is it possible it could return all letters, which would end up getting replaced with all zeros? That wouldn't be ideal. haha Is there a function that just returns random numbers? If not, maybe generate a way longer WordArray to start, so you're much more likely to end up with 6 digits?